### ISSUE TITLE
Add rule to prevent `BigDecimal.valueOf(<some_double_value/some_float_value>)`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6704]
As per the comment at https://github.com/detekt/detekt/issues/6668#issuecomment-1834718270 it carried the issue of precision to `BigDecimal` which might be unintended.

We should create the rule to prevent this

**Comment by BraisGabin** [comment]
I'm not sure if we want a new rule or just add it as a forbidden method call

**Comment by schalkms** [comment]
I agree. For this, we have forbidden method call. @atulgpt might want to add to the default configuration as it concerns the standard Kotlin lib.

**Comment by VirtualParticle** [comment]
`BigDecimal.valueOf(double)` is actually fine to use. Here's the implementation of that method:
```java
public static BigDecimal valueOf(double val) {
    return new BigDecimal(Double.toString(val));
}
```
It's the constructor (`BigDecimal.<init>(double)`) that can produce the unexpected behavior. Here's an excerpt from the JavaDoc for that:
> When a `double` must be used as a source for a `BigDecimal`, note that this constructor provides an exact conversion; it does not give the same result as converting the `double` to a `String` using the `Double.toString(double)` method and then using the `BigDecimal(String)` constructor. To get that result, use the `static` `valueOf(double)` method.

I agree that it's a good idea to add this as a default configuration to the `ForbiddenMethodCall` rule since it's very unlikely someone would actually want that behavior. But it's the constructor we should forbid, not the factory method.
